### PR TITLE
Add section about @QuarkusTestResourse in testing guide

### LINKED ISSUE #9235: Improve the documentation of @QuarkusTestResource and QuarkusTestResourceLifecycleManager
There is pretty few discussion of @QuarkusTestResource and QuarkusTestResourceLifecycleManager in the docs.
I described a concrete usage scenario in https://groups.google.com/d/msg/quarkus-dev/YsmYuKqXkl8/isfUCD3sBwAJ (I don't want to mock the REST client but start a container with it).
The intention is not you present a solution for my problem but give some ideas what is possible.

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #9235

### CODE REVIEW

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 1]
```diff
@@ -573,6 +573,18 @@ public class GreetingResourceTest {
 
 If you are using the `quarkus-hibernate-orm-panache` or `quarkus-mongodb-panache` extensions, check out the link:hibernate-orm-panache#mocking[Hibernate ORM with Panache Mocking] and link:mongodb-panache#mocking[MongoDB with Panache Mocking] documentation for the easiest way to mock your data access.
 
+== Starting services before the Quarkus application starts
+
+A very common need is to start some services on which you Quarkus application depends on, before the Quarkus application starts for testing. To address this need Quarkus provides `@io.quarkus.test.common.QuarkusTestResource` and `io.quarkus.test.common.QuarkusTestResourceLifecycleManager`.
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:578]:
```suggestion
A very common need is to start some services on which your Quarkus application depends, before the Quarkus application starts for testing. To address this need, Quarkus provides `@io.quarkus.test.common.QuarkusTestResource` and `io.quarkus.test.common.QuarkusTestResourceLifecycleManager`.
```

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 2]
```diff
@@ -573,6 +573,18 @@ public class GreetingResourceTest {
...
+
+By simply annotating any test in the test-suite with `@QuarkusTestResource`, Quarkus will run the corresponding `QuarkusTestResourceLifecycleManager` before any tests are run.
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:580]:
```suggestion
By simply annotating any test in the test suite with `@QuarkusTestResource`, Quarkus will run the corresponding `QuarkusTestResourceLifecycleManager` before any tests are run.
```

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 3]
```diff
@@ -573,6 +573,18 @@ public class GreetingResourceTest {
...
+A test-suite is also free to utilize multiple `@QuarkusTestResource` annotations, in which case all the corresponding `QuarkusTestResourceLifecycleManager` objects will be run before the tests.
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:581]:
```suggestion
A test suite is also free to utilize multiple `@QuarkusTestResource` annotations, in which case all the corresponding `QuarkusTestResourceLifecycleManager` objects will be run before the tests.
```

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 4]
```diff
@@ -573,6 +573,18 @@ public class GreetingResourceTest {
...
+
+Quarkus provides a few implementations of `QuarkusTestResourceLifecycleManager` out of the box (see `io.quarkus.test.h2.H2DatabaseTestResource` that starts an H2 database, or `io.quarkus.test.kubernetes.client.KubernetesMockServerTestResource` which starts a mock Kubernetes API server),
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:583]:
```suggestion
Quarkus provides a few implementations of `QuarkusTestResourceLifecycleManager` out of the box (see `io.quarkus.test.h2.H2DatabaseTestResource` which starts an H2 database, or `io.quarkus.test.kubernetes.client.KubernetesMockServerTestResource` which starts a mock Kubernetes API server),
```

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 5]
```diff
@@ -573,6 +573,18 @@ public class GreetingResourceTest {
...
+but it is common to create custom implementations to address specific application needs.
+Common cases include starting docker containers using https://www.testcontainers.org/[testcontainers] (an example of which can be found https://github.com/quarkusio/quarkus-quickstarts/blob/master/kafka-quickstart/src/test/java/org/acme/kafka/KafkaResource.java[here]),
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:585]:
```suggestion
Common cases include starting docker containers using https://www.testcontainers.org/[Testcontainers] (an example of which can be found https://github.com/quarkusio/quarkus-quickstarts/blob/master/kafka-quickstart/src/test/java/org/acme/kafka/KafkaResource.java[here]),
```

File: docs/src/main/asciidoc/getting-started-testing.adoc [hunk 6]
```diff
@@ -573,6 +573,18 @@ public class GreetingResourceTest {
...
+or starting a mock HTTP server using http://wiremock.org/[Wiremock] (an example of which can be found https://github.com/geoand/quarkus-test-demo/blob/master/src/test/java/org/acme/getting/started/country/WiremockCountries.java[here]).
+
```
Comment (gsmet) [inline docs/src/main/asciidoc/getting-started-testing.adoc:587]: You don't want to add a paragraph about `initArgs`?

**Review by gsmet — CHANGES_REQUESTED** [review]
Spotted a couple things.

File: docs/src/main/asciidoc/getting-started-testing.adoc (hunk 6 above)
Comment (geoand) [reply docs/src/main/asciidoc/getting-started-testing.adoc:587]: I think it's too much of a niche feature to be mentioned in the docs. I believe (hope?) the javadoc should be enough :)

File: docs/src/main/asciidoc/getting-started-testing.adoc (hunk 2 above)
Comment (geoand) [reply docs/src/main/asciidoc/getting-started-testing.adoc:580]: Was on the fence about this, but I absolutely trust you on it :)

**Comment by geoand** [comment]
Issues addressed

File: docs/src/main/asciidoc/getting-started-testing.adoc (hunk 6 above)
Comment (geoand) [reply docs/src/main/asciidoc/getting-started-testing.adoc:587]: Resolving for now :)

**Review by gsmet — APPROVED** [review]

**Pull request merged**